### PR TITLE
Better color pallet

### LINKED ISSUE #25: issue#25
Use some better color grading in App for better user experience:

- [x] Repair Sort By to be in one line
- [x] Make better color pallet 
- [x] Search input needs to be dark in  dark theme

### PR DESCRIPTION
**Pull request by SrdjanCukic**
Closes #25

### CODE REVIEW

File: src/UI/ToggleTheme.jsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const SliderToggle = ({ selected, setSelected }) => {
     <div className="relative flex w-fit items-center rounded-full">
       <button
         className={`${TOGGLE_CLASSES} ${
-          selected === 'light' ? 'text-white' : 'text-slate-300'
+          selected === 'light' ? 'text-foreground' : 'text-slate-300'
```
Comment (Ninjaneer87) [inline src/UI/ToggleTheme.jsx]: You don't need this type of check, try just 

```jsx
className={`${TOGGLE_CLASSES} text-foreground`}
```
and see the magic

So, `foreground` has different values for light/dark theme, and they are applied when the themes switch, that's why you have those CSS variables for colors in `body` for light, and `body.dark` for dark

If you want to set something specific for dark mode, for instance, you can do `dark:bg-red-500` and it will apply red background only when it's dark mode

File: src/UI/SearchAndPreferences.jsx [hunk 2]
```diff
@@ -49,7 +50,7 @@ const SearchAndPreferences = () => {
           }}
           placeholder="Search"
           type={'text'}
-          color="secondary"
+          className="bg-transparent"
```
Comment (Ninjaneer87) [inline src/UI/SearchAndPreferences.jsx]: If you're having hard time customizing MUI components, take a look at [this](https://mui.com/material-ui/customization/theme-components/)

TLDR; you can create theme objects, one for light one for dark theme, and configure/customize/override what ever colors/components you wish there

Then you just toggle the themes together with your dark/light existing logic

File: src/components/OnMountApi.jsx [hunk 3]
```diff
@@ -17,27 +17,41 @@ function OnMountApi() {
...
-        {processedDataNewsApi.map((value, index) => (
-          <Article key={`newsapi-${index}`} data={value} />
-        ))}
-        {processedDataGNewsApi.map((value, index) => (
-          <Article key={`gnews-${index}`} data={value} />
-        ))}
-        {processedDataNyTimes.map((value, index) => (
-          <Article key={`nyt-${index}`} data={value} />
+        {allArticles.map(article => (
+          <Article key={article.key} data={article} />
```
Comment (Ninjaneer87) [inline src/components/OnMountApi.jsx]: you can use `article.url` for the key, in this case it is guaranteed to be unique

File: src/components/OnMountApi.jsx [hunk 4]
```diff
@@ -17,27 +17,41 @@ function OnMountApi() {
...
+        ?.filter(
+          item => item.content && item.content !== '[Removed]' && item.url,
+        ) // Ensure the article has content and a valid URL
         ?.slice(0, 3) || [];
 
+    // Combine processed articles
+    const allArticles = [
+      ...processedDataNewsApi.map((article, index) => ({
+        ...article,
+        key: `newsapi-${index}`,
```
Comment (Ninjaneer87) [inline src/components/OnMountApi.jsx]: Ideally, using `index` for `key` should be the last resort

The best way to go is to find a common unique property for the list items, which in this case seems to be `url` property

You can totally use that for the key

Here's a little read about react keys from the docs

https://react.dev/learn/rendering-lists#rules-of-keys

File: src/UI/SearchAndPreferences.jsx [hunk 5]
```diff
@@ -63,7 +63,7 @@ const SearchAndPreferences = () => {
               sm: 'block',
               md: 'block',
               lg: 'block',
-              xl: 'none',
+              xl: 'block',
```
Comment (Ninjaneer87) [inline src/UI/SearchAndPreferences.jsx]: In this case, you don't need to explicitly list (repeat) values for all sizes, you can just do this to achieve the same

If you comment these like in the screenshot, you will get the same thing

image

Comment (Ninjaneer87) [reply src/UI/SearchAndPreferences.jsx]: But there's an even better way, since you're already using tailwind, you can do all this with tailwind classes

A few examples:

image

instead of using `sx` prop, you can add classes to already existing ones in the `className` above, so you could add `flex items-center justify-between` to achieve the same thing

For this next one `ml-1 flex-1`

image


here `p-1 hidden sm:block`

image

and so on, you get the idea, tailwind classes should be able to cover 99% of cases, you'll  rarely ever need to use `sx`
